libs/GLMLoader/GLMLoader.cpp

```diff
@@ -0,0 +1,710 @@
+				v_normals[normalCount++] = normal3;
+			}
+			
+			if (numsUV > 0)
+			{
+				glm::vec2 uv1;
+				uv1.x = texcoords[2 * (triangles[i].txindices[0] - 1) + 0];
+				uv1.y = texcoords[2 * (triangles[i].txindices[0] - 1) + 1];
+				v_textures[textureCount++] = uv1;
+				
+				glm::vec2 uv2;
+				uv2.x = texcoords[2 * (triangles[i].txindices[1] - 1) + 0];
+				uv2.y = texcoords[2 * (triangles[i].txindices[1] - 1) + 1];
+				v_textures[textureCount++] = uv2;
+
+				glm::vec2 uv3;
+				uv3.x = texcoords[2 * (triangles[i].txindices[2] - 1) + 0];
+				uv3.y = texcoords[2 * (triangles[i].txindices[2] - 1) + 1];
+				v_textures[textureCount++] = uv3;
+			}
+
+
+		}
+	
+
+
+
+	free(vertices);
+	free(normals);
+	free(triangles);
+	free(texcoords);
+
+
+
+
+	VertexBuffer* vertexObject = new VertexBuffer();
+
+	vertexObject->setData( v_vertices, v_normals, v_textures, v_indices);
+
+
+	// return model
+	Model* model = new Model();
+	model->setObjModel(vertexObject);
+
+	return model;
+}
+
+Model* GLMLoader::loadObjModel(std::string& filename)
+{
+  tinyobj::ObjReaderConfig reader_config;
+	tinyobj::ObjReader reader;
+  if (!reader.ParseFromFile(filename, reader_config)) {
+    if (!reader.Error().empty()) {
+      std::cerr << "TinyObjReader: " << reader.Error();
+    }
+    exit(1);
+  }
+
+  if (!reader.Warning().empty()) {
+    std::cout << "TinyObjReader: " << reader.Warning();
+  }
+
+  auto& attrib = reader.GetAttrib();
+  auto& shapes = reader.GetShapes();
+  auto& materials = reader.GetMaterials();
+
+	std::vector<float> vertices2 = attrib.GetVertices();
+	//std::vector<float> normals = attrib.normals;
+	std::vector<float> text = attrib.texcoords;
+
+
+  std::vector<glm::vec3> vertices;
+	std::vector<glm::vec2> textcoords;
+	std::vector<glm::vec3> normals;
+  std::vector<unsigned int> indices;
+  std::unordered_map<glm::vec3, int> uniqueVertices;
+	std::unordered_map<glm::vec2, int> uniqueTextCoords;
+	std::unordered_map<glm::vec3, int> uniqueNormals;
+
+  for (const auto& shape : shapes)
+  {
+    for (const auto& index : shape.mesh.indices)
+    {
+      glm::vec3 position{
+          attrib.vertices[3 * index.vertex_index + 0],
+          attrib.vertices[3 * index.vertex_index + 1],
+          attrib.vertices[3 * index.vertex_index + 2] };
+
+        glm::vec2 text_coord{
+         attrib.texcoords[2 * index.texcoord_index + 0],
+         attrib.texcoords[2 * index.texcoord_index + 1]};
+				/*
+        glm::vec3 normal{
+        attrib.texcoords[3 * index.normal_index + 0],
+        attrib.texcoords[3 * index.normal_index + 1],
+        attrib.texcoords[3 * index.normal_index + 2] };*/
+
+      if (uniqueVertices.count(position) == 0)
+      {
+        uniqueVertices[position] = static_cast<int>(vertices.size());
+        vertices.push_back( position );
+      }
+
+      if (uniqueTextCoords.count(text_coord) == 0)
+      {
+				uniqueTextCoords[text_coord] = static_cast<int>(textcoords.size());
```

Comment:
the indentation is a bit inconsistent - is there an auto-formatter these files could be fun through? (like `black` for python or `JuliaFormatter` for julia)

Reply:
following up on my original question, is there a C++ auto formatter that could be used? (make sure indentation is correct and code is prettified)